src/dlfcn.c

```diff
@@ -698,11 +719,28 @@ static BOOL is_import_thunk( const void *addr )
 static void *get_address_from_import_address_table( void *iat, DWORD iat_size, const void *addr )
 {
     BYTE *thkp = (BYTE *) addr;
+#if defined _M_ARM64 || defined __aarch64__
+    /*
+     *  typical import thunk in ARM64:
+     *  0x7ff772ae78c0 <+25760>: adrp   x16, 1
+     *  0x7ff772ae78c4 <+25764>: ldr    x16, [x16, #0xdc0]
+     *  0x7ff772ae78c8 <+25768>: br     x16
+     */
+    ULONG opCode1 = *(ULONG*)((BYTE*)addr);
+    ULONG opCode2 = *(ULONG*)((BYTE*)addr + 4);
+
+    UINT64 pageLow2 = (opCode1 >> 29) & 3;
+    UINT64 pageHigh19 = (opCode1 >> 5) & ~(~0ull << 19);
+    INT64 page = sign_extend((pageHigh19 << 2) | pageLow2, 21) << 12;
+
+    UINT64 offset = ((opCode2 >> 10) & ~(~0ull << 12)) << 3;
+    BYTE* ptr = (BYTE*)((ULONG64)thkp & 0xfffffffffffff000ull) + page + offset;

```

Comment:
```suggestion
    BYTE *ptr = (BYTE *) ( (ULONG64) thkp & 0xfffffffffffff000ull ) + page + offset;
```
or maybe simplify `0xfffffffffffff000ull` by `~0xfffull`